plugins/inputs/gnmi/handler.go

```diff
@@ -98,21 +103,45 @@ func (h *handler) subscribeGNMI(ctx context.Context, acc telegraf.Accumulator, t
 				h.log.Debugf("update_%v: %s", t, string(buf))
 			}
 		}
-
 		if response, ok := reply.Response.(*gnmiLib.SubscribeResponse_Update); ok {
-			h.handleSubscribeResponseUpdate(acc, response)
+			h.handleSubscribeResponseUpdate(acc, response, reply.GetExtension())
 		}
 	}
 	return nil
 }
 
 // Handle SubscribeResponse_Update message from gNMI and parse contained telemetry data
-func (h *handler) handleSubscribeResponseUpdate(acc telegraf.Accumulator, response *gnmiLib.SubscribeResponse_Update) {
+func (h *handler) handleSubscribeResponseUpdate(acc telegraf.Accumulator, response *gnmiLib.SubscribeResponse_Update, extension []*gnmiExt.Extension) {
 	var prefix, prefixAliasPath string
 	grouper := metric.NewSeriesGrouper()
 	timestamp := time.Unix(0, response.Update.Timestamp)
 	prefixTags := make(map[string]string)
 
+	// Check if jnpr_extension option is set
+	if h.jnprExtension {
+		// If extension is present - iter on each extension to find EID_JUNIPER_TELEMETRY_HEADER
+		if len(extension) > 0 {
+			for _, ext := range extension {
+				currentExt := ext.GetRegisteredExt().Msg
+				idExt := ext.GetRegisteredExt().Id
+				// Juniper Telemetry header
+				//EID_JUNIPER_TELEMETRY_HEADER = 1;
+				if currentExt != nil && idExt == 1 {
+					juniperHeader := &jnprHeader.GnmiJuniperTelemetryHeader{}
+					// unmarshal extention
+					result := proto.Unmarshal(currentExt, juniperHeader)
```

Comment:
this should be err not result right?
so the check below should be if err != nil, break
do we want to report the error?